### PR TITLE
[DCJ-198] Refactor NIHDataManagement.js to JSX

### PR DESCRIPTION
**Pull request by PintoGideon**
### Addresses

https://broadworkbench.atlassian.net/browse/DCJ-198

### Summary

1. Remove the use of react-hypescript-helpers
2. Update the file extension from js to jsx

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/data_submission/NIHDataManagement.jsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                    id='otherReasonForRequest'
+                    defaultValue={formData?.alternativeDataSharingPlanReasons?.includes(alternativeDataSharingPlanReasonValues.otherReasonForRequest)}
+                    type={FormFieldTypes.CHECKBOX}
+                    onChange={onAlternativeDataSharingPlanReasonsChange}
+                    toggleText='Other'
+                  />
+                </div>
+              )}
+              <FormField // todo: other reason for request
+                id='alternativeDataSharingPlanExplanation'
```
Comment (rushtong) [inline src/pages/data_submission/NIHDataManagement.jsx]: nit: The comment here looks like it was related to the `isRendered` condition. I'm not sure what it meant in the first place, and less sure what it means here.

**Review by rushtong — APPROVED** [review]
Tested locally, lgtm 👍🏽

**Review by aarohinadkarni — APPROVED** [review]
tested locally looks good 👍

**Review by fboulnois — APPROVED** [review]
Thank you 👍

**Pull request merged**